airflow/providers/microsoft/azure/transfers/local_to_adls.py

```diff
@@ -96,3 +96,53 @@ def execute(self, context: Context) -> None:
             blocksize=self.blocksize,
             **self.extra_upload_options,
         )
+
+
+class DataToADLSOperator(BaseOperator):
```

Comment:
I checked and we have the equivalent operator in S3

https://github.com/apache/airflow/blob/2bc103698fad1b7c0bebed0d5cbdda0ec7ea83c3/airflow/providers/amazon/aws/operators/s3.py#L352


So please lets rename the operator to be `ADLSCreateObjectOperator` and place it in `operators/adsl.py`
Lets also try to keep parameters name according to the conventions we already have in the S3 (where it make sense) for example `replace` rather than `overwrite` this is very helpful for users who work in multi cloud environment. Note also about the extra parameters there are in the S3 example maybe would be good to add them like `encryption`, `encoding`